code/modules/power/smes.dm

```diff
@@ -221,18 +227,60 @@
 
 /obj/machinery/power/smes/update_overlays()
 	. = ..()
-	if(machine_stat & BROKEN)
-		return
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
 
 	if(panel_open)
+		. += "panel"
 		return
 
-	. += "smes-op[outputting ? 1 : 0]"
-	. += "smes-oc[inputting ? 1 : 0]"
+	if(machine_stat & BROKEN)
+		return
 
 	var/clevel = chargedisplay()
-	if(clevel > 0)
-		. += "smes-og[clevel]"
+	if(!terminal)
+		. += "failing"
+		SSvis_overlays.add_vis_overlay(src, icon, "failing", layer, plane, dir)
+		SSvis_overlays.add_vis_overlay(src, icon, "failing", layer, EMISSIVE_PLANE, dir)
```

Comment:
the vis_overlays system is deprecated, has been for years (vis_contents, which is how it works, create higher server load so we do not use them)

Instead, use `mutable_appearance` and `emissive_appearance` (alongside their arguments)
Also, `. += "{icon_state}"` is handled the same as just handing back a mutable appearance with the particular icon state set and all other values inherited from the parent. So these states are being displayed twice incorrectly.

Also of note, emissive_appearance takes an atom to use as a reference for its position in the map (to make multiz work correctly). The smes will likely work just fine for this